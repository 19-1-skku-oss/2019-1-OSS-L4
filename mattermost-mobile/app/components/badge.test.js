// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Text, TouchableWithoutFeedback} from 'react-native';
import {shallow} from 'enzyme';

import Badge from './badge';

describe('Badge', () => {
    const baseProps = {
        count: 1,
        countStyle: {color: '#145dbf', fontSize: 10},
        style: {backgroundColor: '#ffffff'},
        onPress: jest.fn(),
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <Badge {...baseProps}/>
        );

        expect(wrapper.instance().renderText()).toMatchSnapshot();
        expect(wrapper.find(TouchableWithoutFeedback).exists()).toEqual(true);
        expect(wrapper.find(Text).first().props().children).toContain('1');
    });
});
