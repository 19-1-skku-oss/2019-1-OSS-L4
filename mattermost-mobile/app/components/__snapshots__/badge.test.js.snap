// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#444",
        "borderRadius": 20,
        "height": 20,
        "padding": 12,
        "paddingBottom": 3,
        "paddingTop": 3,
        "position": "absolute",
        "right": 30,
        "top": 2,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "opacity": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "textAlignVertical": "center",
        }
      }
    >
      <Text
        onLayout={[Function]}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 14,
            },
            Object {
              "color": "#145dbf",
              "fontSize": 10,
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;
